frontend/src/metabase-lib/lib/queries/StructuredQuery.js

```diff
@@ -1487,42 +1487,38 @@ export default class StructuredQuery extends AtomicQuery {
     return queries;
   }
 
-  dependentTableIds({ includeFKs = true } = {}) {
-    const tableIds = new Set();
+  /**
+   * Metadata this query needs to display correctly
+   */
+  dependentMetadata({ foreignTables = true } = {}) {
```

Comment:
This is tested with the e2e test you added, but there's probably enough logic to justify some metabase-lib unit tests too.

Reply:
Good call, adding some unit tests found a bug :)